src/App.tsx

```diff
@@ -3,7 +3,38 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
+type Movie = {
+  title: string
+  description: string
+  imgUrl: string
+  imdbUrl: string
+  imdbId: string
+};
+
 export const App: React.FC = () => {
+  const [visiblemovies, setVisibleMovies] = React.useState(moviesFromServer);
+  const [query, setQuery] = React.useState('');
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(e.target.value.trim().toLowerCase());
+  };
+
+  React.useEffect(() => {
+    setVisibleMovies((prevMovies = []): Movie[] => {
+      if (!query) {
+        return moviesFromServer;
+      }
+
+      return prevMovies.filter((movie: Movie) => movie.title
+        .toLowerCase().includes(query)
+        || movie.description.toLowerCase().includes(query));
+    });
+  }, [query]);
+
+  React.useEffect(() => {
+    setVisibleMovies(moviesFromServer as Movie[]);
```

Comment:
no need to use these useEffects

you can just create a variable with needed filtered data by query and use it, since you just pass it to a separate component